ote_sdk/ote_sdk/tests/entities/shapes/test_rectangle.py

```diff
@@ -0,0 +1,667 @@
+class TestRectangle:
+    @staticmethod
+    def horizontal_rectangle_params() -> dict:
+        return {"x1": 0.1, "y1": 0.0, "x2": 0.4, "y2": 0.2}
+
+    def horizontal_rectangle(self) -> Rectangle:
+        return Rectangle(**self.horizontal_rectangle_params())
+
+    @staticmethod
+    def vertical_rectangle_params() -> dict:
+        return {
+            "x1": 0.1,
+            "y1": 0.1,
+            "x2": 0.3,
+            "y2": 0.4,
+            "labels": ["label_1", "label_2"],
+            "modification_date": datetime(
+                year=2020, month=1, day=1, hour=9, minute=30, second=15, microsecond=2
+            ),
+        }
+
+    def vertical_rectangle(self) -> Rectangle:
+        return Rectangle(**self.vertical_rectangle_params())
+
+    @staticmethod
+    def square_params() -> dict:
+        return {"x1": 0.1, "y1": 0.1, "x2": 0.3, "y2": 0.3}
+
+    def square(self) -> Rectangle:
+        return Rectangle(**self.square_params())
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_rectangle_required_parameters(self):
+        """
+        <b>Description:</b>
+        Check Rectangle class instance required parameters
+
+        <b>Input data:</b>
+        Rectangle class initiation parameters
+
+        <b>Expected results:</b>
+        Test passes if Rectangle instance has expected attributes specified during required parameters initiation
+
+        <b>Steps</b>
+        1. Compare x1, y1, x2, y2 and type Rectangle instance parameters with expected values
+        """
+        rectangle = self.horizontal_rectangle()
+        assert rectangle.x1 == 0.1
+        assert rectangle.y1 == 0.0
+        assert rectangle.x2 == 0.4
+        assert rectangle.y2 == 0.2
+        assert rectangle.type == ShapeType.RECTANGLE
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_rectangle_optional_parameters(self):
+        """
+        <b>Description:</b>
+        Check Rectangle optional parameters
+
+        <b>Input data:</b>
+        Instance of Rectangle class
+
+        <b>Expected results:</b>
+        Test passes if Rectangle instance has expected labels and modification attributes specified during Rectangle
+        class object initiation with optional parameters
+
+        <b>Steps</b>
+        1. Compare default label Rectangle instance attribute with expected value
+        2. Compare default modification_date Rectangle instance attribute with expected value
+        3. Compare specified label Rectangle instance attribute with expected value
+        4. Compare specified modification_date Rectangle instance attribute with expected value
+        """
+        # check for default values of optional parameters
+        default_params_rectangle = self.horizontal_rectangle()
+        expected_default_params_rectangle_modification_date = now().replace(
```

Comment:
the same issue with `now`

Reply:
I guess that it can be false in some rare cases even if you replace microseconds by 0. 
Imagine that SECOND (but not MICROSECOND) was changed between two calls of `now`. One here and another inside ` self.horizontal_rectangle()`

I would call `now` only once, store it as class member and re-use where it is needed, without replacing microseconds.